battle-engine.js

```diff
@@ -492,6 +492,56 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.getMoveTargets = function (move, target) {
+		move = this.battle.getMove(move);
+		var targets = [];
+		switch (move.target) {
+		case 'all':
+		case 'foeSide':
+		case 'allySide':
+		case 'allyTeam':
+			if (move.target.substr(0, 3) !== 'foe') {
```

Comment:
Style is to use `substr` where substr makes sense and `slice` where slice makes sense.

`substr` makes more sense here.